lib/extractor/layer.ts

```diff
@@ -53,6 +56,73 @@ export async function extractDockerArchive(
   });
 }
 
+export async function extractOciArchive(
+  ociArchiveFilesystemPath: string,
+  extractActions: ExtractAction[],
+): Promise<{
+  layers: ExtractedLayers[];
+  manifest: OciArchiveManifest;
+}> {
+  return new Promise((resolve, reject) => {
+    const tarExtractor: Extract = extract();
+
+    const layers: Record<string, ExtractedLayers> = {};
+    const manifests: Record<string, OciArchiveManifest> = {};
+    let imageIndex: OciImageIndex | undefined;
+
+    tarExtractor.on("entry", async (header, stream, next) => {
+      if (header.type === "file") {
+        if (isImageIndexFile(header.name)) {
+          imageIndex = await streamToJson<OciImageIndex>(stream);
+        } else {
+          const jsonStream = new PassThrough();
+          const layerStream = new PassThrough();
+          stream.pipe(jsonStream);
+          stream.pipe(layerStream);
+
+          const promises = [
+            streamToJson(jsonStream).catch(() => undefined),
```

Comment:
why are we okay with disregarding errors here?

Reply:
yes, exactly, if we get an error, we were parsing a layer, so we don't care about the error

Reply:
I think I didn't say error enough times!